lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -82,6 +82,24 @@ CrossOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
   return success();
 }
 
+LogicalResult UnionDistinctOp::inferReturnTypes(
+    MLIRContext *context, std::optional<Location> loc, ValueRange operands,
+    DictionaryAttr attributes, OpaqueProperties properties, RegionRange regions,
+    llvm::SmallVectorImpl<Type> &inferredReturnTypes) {
+
+  Value leftInput = operands[0];
+  Value rightInput = operands[1];
+
+  TypeRange leftFieldTypes = cast<TupleType>(leftInput.getType()).getTypes();
+  TypeRange rightFieldTypes = cast<TupleType>(rightInput.getType()).getTypes();
```

Comment:
Oh, I am sorry, I think this got lost during rebase: Wasn't there a check for matching types here? I hope you can still see it in your original git folder using `git show f2b0c24`.

Reply:
Also, if you add that check back, can you add a test for invalid IR that produces that error message (like in [this](https://github.com/substrait-io/substrait-mlir-contrib/blob/main/test/Dialect/Substrait/emit-invalid.mlir) test)?

Reply:
I've added everything now - it should work now! Sorry for the many commits!

Reply:
Awesome, thanks!